content/source/docs/enterprise/private/minio-setup-guide.html.md

```diff
@@ -0,0 +1,95 @@
+sidebar_current: "docs-enterprise2-private-installer-minio"
+description: |-
+  This document provides an overview for setting up minio for external object storage for HashiCop Private Terraform Enterprise (PTFE).
+---
+
+## Introduction
+
+This document provides an overview for setting up minio for external object storage for HashiCop Private Terraform Enterprise (PTFE).
+
+When configured to use external services, PTFE must be connected to a storage service to persist workspace state and other file-based data.  Native support exists for Azure Blob Storage, Amazon S3, and services that are API-compatible with Amazon S3.  If you are not using Azure or a cloud provider with an S3-compatible service, or you are running PTFE in an environment without a storage service, it may be possible to use [minio](https://minio.io) instead.  Minio is an Amazon S3-compatible object storage server that can use mounted filesystems or a variety of other cloud storage solutions to persist data.  HashiCorp may add support in the future for other cloud storage solutions to meet customer requests.
+
+## Required Reading
+
+- Ensure you are familiar with PTFE's operation and [installation requirements](./installer.html), and especially the Operational Mode Decision.
+- Familiarize yourself with [minio](https://minio.io).
+
+## Overview
+
+This guide will walk through installing minio in a Docker container alongside PTFE on the same host, with PTFE configured in the "Production - External Services" [operational mode](.install-installer.html#operational-mode-decision).  This is not a production-ready configuration: it's intended to guide you to a working configuration that can later be automated and hardened.  Data will not be persisted outside of an ephemeral Docker volume, minio will not start on system boot, etc.  It is assumed your instance will have access to the Internet and that you will be performing an online install of PTFE.
+
+## Installation
+
+### System preparation
+
+Ensure your Linux instance meets the [requirements](./install-installer.html#linux-instance).  You will need [jq](https://stedolan.github.io/jq/) and the [AWS cli](https://aws.amazon.com/cli/).
+
+You also need a PostgreSQL database that meets the [requirements](.install-installer.html#postgresql-requirements), as this is part of the external services operational mode.
+
+### PTFE installation
+
+Begin with an [online installation](./install-installer.html#run-the-installer-online).  Once the installation script has finished and you're presented with the following text, move on to the next section:
+
+```
+To continue the installation, visit the following URL in your browser:
+
+  https://<this_server_address>:8800
+```
+
+### Start minio
+
+Now you'll start the minio container, mounting a volume so that you can gain access to the generated config:
+
+    docker run \
+      -d \
+      --name minio \
+      -v /run/minio/config:/root/.minio minio/minio:latest \
```

Comment:
If you break out the rest of the arguments pre line, you should break out the image too.